docs/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,141 @@
+For instance, all classes support the `take` method, so those benchmarks belong in `API/bench_frame_or_index.py`.
+
+```{note}
+`pytest` does not support having two benchmark files with the same name, even if they are in separate directories.
+Therefore, benchmarks of internal methods of _public_ classes go in files suffixed with `_internal`.
+Benchmarks of `GeoDataFrame.polygons.xy`, for instance, belong in `internal/bench_geodataframe_internal.py`.
+```
+
+Free functions have more flexibility.
+Broadly speaking, they should be grouped into benchmark files containing similar functionality.
+For example, I/O benchmarks can all live in `io/bench_io.py`.
+For now those groupings are left to the discretion of developers.
+
+## Running benchmarks
+
+By default, pytest discovers test files and functions prefixed with `test_`.
+For benchmarks, we configure `pytest` to instead search using the `bench_` prefix.
+After installing `pytest-benchmark`, running benchmarks is as simple as just running `pytest`.
+
+When benchmarks are run, the default behavior is to output the results in a table to the terminal.
+A common requirement is to then compare the performance of benchmarks before and after a change.
+We can generate these comparisons by saving the output using the `--benchmark-autosave` option to pytest.
+When using this option, after the benchmarks are run the output will contain a line:
+```
+Saved benchmark data in: /path/to/XXXX_*.json
+```
+
+The `XXXX` is a four-digit number identifying the benchmark.
+If preferred, a user may also use the `--benchmark-save=NAME` option,
+which allows more control over the resulting filename.
+Given two benchmark runs `XXXX` and `YYYY`, benchmarks can then be compared using
+```
+pytest-benchmark compare XXXX YYYY
+```
+Note that the comparison uses the `pytest-benchmark` command rather than the `pytest` command.
+`pytest-benchmark` has a number of additional options that can be used to customize the output.
+The next line contains one useful example, but developers should experiment to find a useful output
+```
+pytest-benchmark compare XXXX YYYY --sort="name" --columns=Mean --name=short --group-by=param
+```
+
+For more details, see the [`pytest-benchmark` documentation](https://pytest-benchmark.readthedocs.io/en/latest/comparing.html).
+
+## Benchmark contents
+
+### Benchmark configuration
+
+Benchmarks must support [comparing to geopandas](#comparing-to-geopandas) and [being run as tests](#testing-benchmarks).
+To satisfy these requirements, one must follow these rules when writing benchmarks:
+1. Import `cuspatial` and `cupy` from the config module:
+   ```python
+       from ..common.config import cuspatial  # Do this
+       import cuspatial  # Not this
+   ```
+   This enables swapping out for `geopandas`.
+2. Avoid hard-coding benchmark dataset sizes, and instead use the sizes advertised by `config.py`.
+   This enables running the benchmarks in "test" mode on small datasets, which will be much faster.
```

Comment:
This too. Although I think the idea can temporarily be extended to that we encourage user to reuse the geodataframes predefined in `conftest.py` instead of hard coding sizes.